controllers/openstackipset_controller.go

```diff
@@ -246,6 +245,199 @@ func (r *OpenStackIPSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 func (r *OpenStackIPSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ospdirectorv1beta1.OpenStackIPSet{}).
-		//		Owns(&corev1.ConfigMap{}).
 		Complete(r)
 }
+
+func (r *OpenStackIPSetReconciler) removeIP(instance *ospdirectorv1beta1.OpenStackIPSet, removedHosts []string) {
+
+	// remove entry from OSIPSet status
+	for _, hostname := range removedHosts {
+		delete(instance.Status.HostIPs, hostname)
+		r.Log.Info(fmt.Sprintf("OSP Hostname %s removed from OSIPSet %s, new status %v", hostname, instance.Name, instance.Status.HostIPs))
+	}
+
+}
+
+func (r *OpenStackIPSetReconciler) syncDeletedIPs(instance *ospdirectorv1beta1.OpenStackIPSet) error {
+
+	allActiveIPSetHosts := make([]string, len(instance.Status.HostIPs))
+	for hostname := range instance.Status.HostIPs {
+		allActiveIPSetHosts = append(allActiveIPSetHosts, hostname)
+	}
+
+	// mark entry in OSNet status as deleted
+	for _, netName := range instance.Spec.Networks {
+		network := &ospdirectorv1beta1.OpenStackNet{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: netName, Namespace: instance.Namespace}, network)
+		if err != nil {
+			if k8s_errors.IsNotFound(err) {
+				r.Log.Info(fmt.Sprintf("OpenStackNet named %s not found!", netName))
```

Comment:
Do we maybe want to `continue` here?  Just wondering if you meant to do something in addition to printing the error message.  It will just return the err, but that could be what you want.

Reply:
yes, just wanted to record the missing network and continue as we don't have to sync deleted ips in that case. I'll add continue